crates/mir/src/emit/emitter.rs

```diff
@@ -2,35 +2,56 @@ use std::{mem, ops::Range};
 
 use rokugo_backend_common::{FunctionId, ValueId};
 
-use super::{content::MirContent, op_code::MirOpCode};
+use super::{
+    container::{MirContainer, MirContainerIterator},
+    op_code::{MirInstruction, MirOpCode},
+};
 
 #[derive(Debug)]
 pub struct MirEmitter {
     next_value_id: u32,
-    content: MirContent,
+    content: MirContainer,
 }
 
 impl MirEmitter {
     pub fn new() -> Self {
         Self {
             next_value_id: 0,
-            content: MirContent { data: Vec::new() },
+            content: MirContainer { data: Vec::new() },
         }
     }
+
+    pub fn iter(&self) -> MirContainerIterator {
+        self.content.iter()
+    }
```

Comment:
Can't we make `content` public instead of doing this? Pretty sure the emitted MIR should be fine to access from the outside.

Reply:
Two functions should exists, because why not?

Reply:
It's just about not repeating yourself. If you're emitting the MIR, you're gonna need to extract it out of the `MirEmitter` later anyways - the simplest way to enable that is to make `content` public, and then the `iter` function on the `MirEmitter` is unnecessary.

Also think about API boundaries - a MIR *emitter* should probably not concern itself with *reading* MIR, should it?